src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -2208,9 +2227,12 @@ void RadioSpinel<InterfaceType, ProcessContextType>::RecoverFromRcpFailure(void)
     if (mResetRadioOnStartup)
     {
         SuccessOrDie(SendReset());
+        SuccessOrDie(WaitResponse());
+    }
+    else
+    {
+        mIsReady = true;
     }
```

Comment:
The above change allows recovery without resetting the RCP, which is necessary in the case of multiple host processes.